megamek/src/megamek/client/ui/swing/MiniMap.java

```diff
@@ -731,148 +516,123 @@ protected synchronized void drawMapOrig() {
-                            dir = m_client.getLocalPlayer().getStartingPos();
-                        }
-
-                        for (int j = 0; j < m_board.getWidth(); j++) {
-                            for (int k = 0; k < m_board.getHeight(); k++) {
-                                if (m_board.isLegalDeployment(
-                                        new Coords(j, k), dir)) {
-                                    paintSingleCoordBorder(g, j, k,
-                                                           Color.yellow);
-                                }
-                            }
-                        }
-                    }
-                }
-                
+            drawDeploymentZone(g);
+            
+            if ((null != game)) {
                 // draw declared fire
-                if ((IGame.Phase.PHASE_FIRING == m_game.getPhase())
-                    || (IGame.Phase.PHASE_PHYSICAL == m_game.getPhase())) {
-                    for (Enumeration<EntityAction> iter = m_game.getActions(); iter
-                            .hasMoreElements(); ) {
-                        EntityAction action = iter.nextElement();
-                        if (action instanceof AttackAction) {
-                            paintAttack(g, (AttackAction) action);
-                        }
+                for (var action : game.getActionsVector()) {
+                    if (action instanceof AttackAction) {
+                        paintAttack(g, (AttackAction) action);
                     }
                 }
 
                 multiUnits.clear();
-                for (Entity e : m_game.getEntitiesVector()) {
-                    if (e.getPosition() == null) {
-                        continue;
+                for (Entity e : game.getEntitiesVector()) {
+                    if (e.getPosition() != null) {
+                        paintUnit(g, e);
                     }
-                    paintUnit(g, e);
                 }
             }
-            clean();
-        }
+            
 
-        if ((m_client != null) && (m_client.getArtilleryAutoHit() != null)) {
-            for (int i = 0; i < m_client.getArtilleryAutoHit().size(); i++) {
-                drawAutoHit(g, m_client.getArtilleryAutoHit().get(i));
+            if ((client != null) && (client.getArtilleryAutoHit() != null)) {
+                for (int i = 0; i < client.getArtilleryAutoHit().size(); i++) {
+                    drawAutoHit(g, client.getArtilleryAutoHit().get(i));
+                }
             }
         }
 
-        if (m_dialog instanceof JDialog) {
-            drawBtn(g);
+        if (dialog != null) {
+            drawButtons(g);
+            repaint();
         }
+    }
+    
+    /** Indicates the deployment hexes. */
+    private void drawDeploymentZone(Graphics g) {
+        if ((null != client) && (null != game) 
+                && IGame.Phase.PHASE_DEPLOYMENT == game.getPhase() && dialog != null) {
```

Comment:
When in Rome, I guess:
```suggestion
                && (IGame.Phase.PHASE_DEPLOYMENT == game.getPhase()) && (dialog != null)) {
```